cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/SpeakerEmotionInference.scala

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.cognitive
+
+import com.microsoft.azure.synapse.ml.cognitive.TextToSpeechSSMLFormat.TextToSpeechSSMLResponseFormat
+import com.microsoft.azure.synapse.ml.logging.BasicLogging
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import com.microsoft.azure.synapse.ml.stages.Lambda
+import org.apache.http.client.methods.HttpRequestBase
+import org.apache.http.entity.{AbstractHttpEntity, StringEntity}
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.ml.{ComplexParamsReadable, NamespaceInjections, PipelineModel, Transformer}
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.types.StructType
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json._
+
+trait HasLocaleCol extends HasServiceParams {
+  val locale = new ServiceParam[String](this,
+    "locale",
+    s"The locale of the input text",
+    isRequired = true)
+
+  def setLocale(v: String): this.type = setScalarParam(locale, v)
+
+  def setLocaleCol(v: String): this.type = setVectorParam(locale, v)
+}
+
+trait HasVoiceCol extends HasServiceParams {
+  val voice = new ServiceParam[String](this,
+    "voice",
+    s"The name of the voice used for synthesis",
+    isRequired = true)
+
+  def setVoiceName(v: String): this.type = setScalarParam(voice, v)
+
+  def setVoiceNameCol(v: String): this.type = setVectorParam(voice, v)
+}
+
+trait HasInput extends HasServiceParams {
+  val input = new ServiceParam[String](this,
+    "input",
+    s"The text input to annotate with inferred emotion")
+
+  def setInput(v: String): this.type = setScalarParam(input, v)
+
+  def setInputCol(v: String): this.type = setVectorParam(input, v)
+}
+
+trait HasSpeakerEmotionInferenceParams extends HasServiceParams
+  with HasLocaleCol with HasVoiceCol with HasSetLocation
+  with HasInput
+
+object SpeakerEmotionInference
+  extends ComplexParamsReadable[SpeakerEmotionInference] with Serializable
+
+class SpeakerEmotionInference(override val uid: String) extends CognitiveServicesBase(uid)
+  with HasSpeakerEmotionInferenceParams with HasCognitiveServiceInput
+  with HasInternalStringOutputParser
+  with BasicLogging {
```

Comment:
Does this extend HasOutputCol somewhere in the stack? If not it should so that people can re-wire the output of this transformer to land in any column they want

Reply:
Yep, it's in CognitiveServicesBaseNoHandler